File: client/src/types/accesorios.ts
export type Accesorio = {
  id: number
  tipo: string
  nombre: string
  color: string
  descripcion: string
  estado: boolean
  id_producto: number
  /** Unidades disponibles (viene de `productos.stock`). */
  stock: number
  foto_url?: string | null
}

export type AccesorioCreatePayload = {
  tipo: string
  nombre: string
  color: string
  descripcion: string
  precio: number
  estado: boolean
  stock?: number
}

export type AccesorioPatchPayload = Partial<AccesorioCreatePayload> & {
  id_producto?: number | null
}

File: src/app/models/accesorios.py
from app.db import Base
from sqlalchemy import Column, Boolean, Integer, String, DateTime, ForeignKey, Text, Numeric, Boolean
from sqlalchemy.orm import relationship


class Accesorios(Base):
    __tablename__="accesorios"

    id = Column(Integer,primary_key=True,autoincrement=True)
    tipo = Column(String(225),nullable=False)
    nombre = Column(String(225),nullable=False)
    color = Column(String(225),nullable=False)
    descripcion = Column(String(225),nullable=False)
    foto_url = Column(String(255), nullable=True)
    estado = Column(Boolean,default=True)
    id_producto = Column(Integer,ForeignKey("productos.id"),nullable=False)

    #relaciones
    productos = relationship("Productos", back_populates="accesorios")
